libeos-updater-util/flatpak-util.c

```diff
@@ -1356,26 +1430,12 @@ squash_ref_actions_ptr_array (GPtrArray *ref_actions  /* (element-type EuuFlatpa
   for (i = 0; i < ref_actions->len; ++i)
     {
       EuuFlatpakRemoteRefAction *action = g_ptr_array_index (ref_actions, i);
-      EuuFlatpakRemoteRefAction *existing_action_for_ref = NULL;
-
-      /* A little trickier than just blindly replacing, there are special
-       * rules regarding "update" since it only updates an existing installed
-       * flatpak, as opposed to installing it.
-       *
-       * (1) "install" and "uninstall" always take priority over "update"
-       *     since "install" means "install or update" and "uninstall"
-       *     means "unconditionally remove".
-       * (2) "update" does not take priority over "install" or "uninstall",
-       *     since the former would subsume it anyway and the latter would
-       *     make the app no longer be installed in that run of the flatpak
-       *     installer.
-       */
-      existing_action_for_ref = g_hash_table_lookup (hash_table, action->ref->ref);
+      EuuFlatpakRemoteRefAction *existing_action_for_ref =
+        g_hash_table_lookup (hash_table, action->ref->ref);
 
-      if (action->type == EUU_FLATPAK_REMOTE_REF_ACTION_INSTALL ||
-          action->type == EUU_FLATPAK_REMOTE_REF_ACTION_UNINSTALL ||
-          existing_action_for_ref == NULL ||
-          existing_action_for_ref->type == EUU_FLATPAK_REMOTE_REF_ACTION_UPDATE)
+      if (existing_action_for_ref == NULL ||
+          (ref_action_type_priority (action->type) >=
+           ref_action_type_priority (existing_action_for_ref->type)))
```

Comment:
aiui this logic will squash `[uninstall org.gnome.Foobar, prune-ref org.gnome.Foobar]` into just `[uninstall org.gnome.Foobar]`, dropping the prune-ref. Is that what we want? Are we sure that uninstalls will now always prune?